src/main/java/com/exactpro/th2/fix/client/Main.java

```diff
@@ -221,15 +229,15 @@ public static void run(Settings settings, MessageRouter<MessageGroupBatch> messa
                             LOGGER.error("Logon rejected, message not sent");
 
                             EventID eventID = message.getMessage().getParentEventId();
-                            String parentEventID = eventID.getId();
+                            String parentEventID = eventID != null ? eventID.getId() : rootEventID;
```

Comment:
`eventID` will never be `null` - as I've said we should check that its `id` is not empty